lib/reverse_sentence.rb

```diff
@@ -1,6 +1,20 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
+def split_my_sentence(sentence)
+  sentence.scan(/\S+|\s/)
+end
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  if my_sentence != nil || my_sentence == ""
+    array = split_my_sentence(my_sentence)
+
+    my_sentence.clear
+
+    array.each do |element|
+      my_sentence.prepend(element)
```

Comment:
Note:  `.prepend` is an O(n) operation all by itself.  This is because each subsequent element in the array has to shift over one index.